### ISSUE TITLE
Versioned docs

### ISSUE THREAD

**Issue by jni** [napari/napari#1385]
## 📚 Documentation

This ties in with #1226. We should have a versioned webpage with docs for each release of napari. Some examples of where this is implemented:

https://docs.bokeh.org/en/latest/index.html
https://scikit-image.org/docs/stable/
https://scikit-learn.org/dev/versions.html

Some guides for achieving this:

- ReadTheDocs's documentation on versioned docs: https://docs.readthedocs.io/en/stable/versions.html
- sphinxcontrib-versioning, a sphinx plugin for this: https://sphinxcontrib-versioning.readthedocs.io/en/latest/, though it seems we might need the github master: https://github.com/sphinx-contrib/sphinxcontrib-versioning

**Comment by kne42** [comment]
Ok so unfortunately `sphinxcontrib-versioning` doesn't work and hasn't been updated in months. I've also tried using `sphinx-multiversion` but it keeps stumbling into errors on my machine. I've made an issue to report it but the developer hasn't seemed to reply to things for 23 days. I'm opposed to using RTD for hosting.

There's a few options we can take from here:
1. Make a script to generate docs for all past versions using `git checkout` (hard)
2. Add to our CI a script that will build documentation and publish it whenever we make a new release, starting with our next release (easy)

I'm going to proceed with (2) for now as it's significantly easier to do and our docs haven't exactly been clean enough in the past for it to be worthwhile building them in addition to our newer docs.
